dns-server/tests/basic_test.rs

```diff
@@ -479,3 +562,36 @@ async fn dns_records_list(
         .map(|z| z.records)
         .unwrap_or_else(HashMap::new))
 }
+
+/// issue a DNS query of `record_ty` records for `name`.
+///
+/// in most tests we just use `trust-dns-resolver` for a higher-level interface
+/// to making DNS queries and handling responses. this is also the crate used
+/// elsewhere to handle DNS responses in Omicron. however, it is slightly
+/// higher-level than the actual wire response we produce from the DNS server in
+/// this same crate; to assert on responses we send on the wire, this issues a
+/// query using `trust-dns-client` and returns the corresponding `DnsResponse`.
+// FIXME(ixi): while `trust-dns-client` has an `AsyncClient` which theoretically
+// integrates nicely with tokio use in this test, I can't figure out how to
+// instantiate it! `SyncClient` is just as good but makes for a slightly messier
+// test.
```

Comment:
i think you can get it from [here](https://docs.rs/trust-dns-client/latest/trust_dns_client/client/struct.AsyncClient.html#method.new), or am i missing the point?

Reply:
i think i hadn't realized how to get a [BufDnsStreamHandle](https://docs.rs/trust-dns-proto/0.23.2/trust_dns_proto/xfer/struct.BufDnsStreamHandle.html), but `BufDnsStreamHandle::new()` does seem like the right thing. i'll give it a try!

Reply:
oh, no, it was `stream` that i couldn't figure out. [UdpClientConnect](https://docs.rs/trust-dns-proto/0.23.2/trust_dns_proto/udp/struct.UdpClientConnect.html) would be right, which you can get from [UdpClientStream](https://docs.rs/trust-dns-proto/0.23.2/trust_dns_proto/udp/struct.UdpClientStream.html#method.new), but:
* the docs on `UdpClientStream::new` mention returning a tuple of the stream and a handle to send messages while it definitely does _not_ return a tuple,
* `BufDnsStreamHandle::new` alleges it takes a `sender` argument to send data, but does not (seems like the missing handle from `UdpClientStream::new`?)

and then i wasn't really sure what to do with `BufDnsStreamHandle::new`'s returned `StreamReceiver`. i also couldn't quickly figure out how to use `AsyncClient` by reading through hickory-dns' resolver.

that's about the point that i looked at `SyncClient` and decided it's good enough for these tests :grimacing:

Reply:
it turns out [`AsyncClient::connect`](https://docs.rs/hickory-client/0.24.1/hickory_client/client/struct.AsyncClient.html#method.connect) side-steps all that and is quite straightforward. i'll get this moved over. (noticed while looking at [tangentially-related tests](https://github.com/hickory-dns/hickory-dns/blob/main/tests/integration-tests/tests/truncation_tests.rs#L24-L36))

Reply:
uses AsyncClient now, as of https://github.com/oxidecomputer/omicron/pull/6308/commits/3f754ee4bd170921669f27a9a9a59f8d31ad2812. thanks for the nudge!